### PR TITLE
Add staticcheck linter

### PR DESCRIPTION
**Pull request by syscll**
Based on the comments in #30, this PR adds the `staticcheck` linter and fixes errors found during analysis.

### CODE REVIEW

**Comment by Foxboron** [comment]
Can you squash the commits?

**Comment by syscll** [comment]
I think it's easier to squash and merge?

**Comment by syscll** [comment]
Settings -> Merge Button -> Allow squash merging

**Comment by Foxboron** [comment]
I don't merge through github as It does faux signatures on the commits. I can manually pull and squash if it's easier :) Ill be doing it for the merge regardless.

**Comment by syscll** [comment]
Ah, no worries. I don't have access to my laptop atm, so if you wouldn't mind then I'd appreciate it!

File: README.md [hunk 1]
```diff
@@ -1,5 +1,4 @@
-sbctl - Secure Boot Manager
-===========================
+# sbctl - Secure Boot Manager [![Build Status](https://github.com/Foxboron/sbctl/workflows/build/badge.svg)](https://github.com/Foxboron/sbctl/actions)
```
Comment (ericonr) [inline README.md:1]: I think it's missing a new line here, isn't it?

File: go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module github.com/foxboron/sbctl
 
-go 1.14
+go 1.15
```
Comment (ericonr) [inline go.mod:3]: I'm not sure what the effect of increasing this is...

File: Makefile [hunk 3]
```diff
@@ -47,3 +47,17 @@ push-aur:
...
 	rm -f sbctl
+
+.PHONY: deps
+deps:
+	go get honnef.co/go/tools/cmd/staticcheck@2020.1.6
+
+.PHONY: lint
+lint: deps
+	go vet ./...
+	test -z $(shell go fmt ./...)
```
Comment (ericonr) [inline Makefile:58]: You might want to quote this, I'm not sure how it behaves in make.

File: Makefile [hunk 4]
```diff
@@ -47,3 +47,17 @@ push-aur:
 clean:
 	rm -f $(MANS)
 	rm -f sbctl
+
+.PHONY: deps
+deps:
+	go get honnef.co/go/tools/cmd/staticcheck@2020.1.6
+
+.PHONY: lint
+lint: deps
```
Comment (ericonr) [inline Makefile:56]: I would prefer to be able to obtain `staticcheck` on my own; perhaps CI should do `make deps lint` for the dependency, but a user can obtain in in other ways.

File: README.md (hunk 1 above)
Comment (syscll) [reply README.md:1]: You would prefer the build status on a new line?

File: go.mod (hunk 2 above)
Comment (syscll) [reply go.mod:3]: I agree, will revert.

File: Makefile (hunk 3 above)
Comment (syscll) [reply Makefile:58]: I'm not actually sure this is the best method of doing this, so I might revert.

File: Makefile (hunk 4 above)
Comment (syscll) [reply Makefile:56]: I would have to disagree here as that I think linting tools should be versioned just like dependencies. That being said, I can remove the call to `deps` when running `lint`, that way you can still use your own version.

File: Makefile (hunk 3 above)
Comment (syscll) [reply Makefile:58]: `go fmt` annoyingly doesn't return error codes.

**Comment by syscll** [comment]
PTAL

File: Makefile (hunk 4 above)
Comment (ericonr) [reply Makefile]: I don't disagree with the versioning, definitely. That sounds good to me.

File: README.md (hunk 1 above)
Comment (ericonr) [reply README.md]: I think so, @Foxboron might disagree with me.

**Comment by syscll** [comment]
Replaced by #35

**Pull request closed without merging**